test/sim/misc/endlessbattleclause.js

```diff
@@ -98,4 +98,23 @@ describe('Endless Battle Clause (slow)', () => {
 		assert(battle.ended);
 		assert(battle.winner === 'Player 2');
 	});
+
+	it('should allow for a maximum of 1000 turns', function () {
+		this.timeout(0);
+		battle = common.createBattle({endlessBattleClause: true});
```

Comment:
Switching this to `common.gen(2).createBattle` causes the test to fail - I believe this reveals there is a bug in how custom rule parsing is handled (ie. `"gen2customgame@@@!Team Preview,!Cancel Mod,Endless Battle Clause"`), but after an hour I gave up trying to track it down since its out of scope.

Reply:
Weird. I feel like I've worked on this code recently enough that I have a pretty good handle on how it works, and I'm very surprised this kind of bug is still here. I'll have to investigate when I have time.